### DISCUSSION TITLE
Bit confused about GameStateTest.java Get Score Assert

### DISCUSSION THREAD

**Discussion by DelusionArun** [CastleLab/COMP3021-2021Fall-PA1-Student-Version#11] [Q&A] [answered]
In line 233-249 of GameStateTest.java, I think the test creates a 2x2 board and adds one gem. So, should the score not be 14 (2 * 2+1 * 10) instead of 4? I am confused over why is it 4?

**Comment by Derppening** [comment, marked as answer]
You only get 10 points *after* you pick up the gem. The gem existing on the board doesn't contribute to the score.

**Comment by DelusionArun** [comment]
Thank you for clarifying.
 
Another question I have is about HasLost() of GameState.java. The documentation says that "For games which the player has unlimited lives, this condition should never trigger."
Does it imply that the function will never be called for unlimited lives or that we need to throw an exception when it does?

**Reply by Derppening** [reply]
It means that for games with unlimited lives, the game will never be lost.
